esp-wifi/src/wifi/mod.rs

```diff
@@ -262,41 +276,94 @@ pub struct AccessPointConfiguration {
     pub secondary_channel: Option<u8>,
 
     /// The set of protocols supported by the access point.
-    #[cfg_attr(feature = "defmt", defmt(Debug2Format))]
-    pub protocols: EnumSet<Protocol>,
+    pub protocols: ProtocolSet,
```

Comment:
How is the user supposed to use this type? They can print it, compare it, but can't really do much more, I think?

Reply:
I actually thought about what to offer - I thought about a very small subset of `EnumSet` (just delegating to it) but wasn't sure if and what makes sense

I'm very open to suggestions here

Reply:
Since we currently have a manual implementation for `defmt::Format`, maybe we can turn this field back into an `EnumSet`, and wrap it into ProtocolSet in the Format implementation only.

Reply:
Hold on

Reply:
Ah too bad, `defmt(ProtocolSet)` is not possible at this point :(

Reply:
I mean we can just remove it from `AccessPointInfo` since we don't know where to get that info from 😆

Reply:
I leave that part to you, but my idea of wrapping the EnumSet is blocked on https://github.com/knurling-rs/defmt/issues/502 - so let me just link that issue here :)

Reply:
👍 I will remove it - that info is not available and probably never will